src/test/java/seedu/rc4hdb/logic/commands/modelcommands/FilterCommandTest.java

```diff
@@ -0,0 +1,127 @@
+        assertEquals(Arrays.asList(ALICE), model.getFilteredResidentList());
+    }
+
+    @Test
+    public void execute_someFieldsSpecifiedUnfilteredList_success() {
+        ResidentDescriptor descriptor = new ResidentDescriptorBuilder().withName(ALICE.getName().toString())
+                .withPhone(ALICE.getPhone().toString()).build();
+        String expectedMessage = String.format(MESSAGE_RESIDENTS_LISTED_OVERVIEW, 1);
+        AttributesMatchKeywordsPredicate predicate =
+                new AttributesMatchKeywordsPredicate(descriptor);
+        FilterCommand command = new FilterCommand(descriptor);
+        expectedModel.updateFilteredResidentList(predicate);
+        assertCommandSuccess(command, model, expectedMessage, expectedModel);
+        assertEquals(Arrays.asList(ALICE), model.getFilteredResidentList());
+    }
+
+    @Test
+    public void execute_nameSpecifiedUnfilteredList_success() {
+        ResidentDescriptor descriptor = new ResidentDescriptorBuilder().withName(ALICE.getName().toString()).build();
+        String expectedMessage = String.format(MESSAGE_RESIDENTS_LISTED_OVERVIEW, 1);
+        AttributesMatchKeywordsPredicate predicate =
+                new AttributesMatchKeywordsPredicate(descriptor);
+        FilterCommand command = new FilterCommand(descriptor);
+        expectedModel.updateFilteredResidentList(predicate);
+        assertCommandSuccess(command, model, expectedMessage, expectedModel);
+        assertEquals(Arrays.asList(ALICE), model.getFilteredResidentList());
+    }
+
+    @Test
+    public void execute_phoneSpecifiedUnfilteredList_success() {
+        ResidentDescriptor descriptor = new ResidentDescriptorBuilder().withPhone(ALICE.getPhone().toString()).build();
+        String expectedMessage = String.format(MESSAGE_RESIDENTS_LISTED_OVERVIEW, 1);
+        AttributesMatchKeywordsPredicate predicate =
+                new AttributesMatchKeywordsPredicate(descriptor);
+        FilterCommand command = new FilterCommand(descriptor);
+        expectedModel.updateFilteredResidentList(predicate);
+        assertCommandSuccess(command, model, expectedMessage, expectedModel);
+        assertEquals(Arrays.asList(ALICE), model.getFilteredResidentList());
+    }
+
+    @Test
+    public void execute_emailSpecifiedUnfilteredList_success() {
+        ResidentDescriptor descriptor = new ResidentDescriptorBuilder().withEmail(ALICE.getEmail().toString()).build();
+        String expectedMessage = String.format(MESSAGE_RESIDENTS_LISTED_OVERVIEW, 1);
+        AttributesMatchKeywordsPredicate predicate =
+                new AttributesMatchKeywordsPredicate(descriptor);
+        FilterCommand command = new FilterCommand(descriptor);
+        expectedModel.updateFilteredResidentList(predicate);
+        assertCommandSuccess(command, model, expectedMessage, expectedModel);
+        assertEquals(Arrays.asList(ALICE), model.getFilteredResidentList());
+    }
+
```

Comment:
I was expecting `Gender` and `House` tests in between here hahaha.

Might be good to add them, in case a future change breaks the `Gender` or `House` filter.